tests/DynamoDbCompositeModelTest.php

```diff
@@ -42,6 +42,30 @@ public function testCreateRecord()
         $this->assertEquals($this->testModel->id2, $record['Item']['id2']['S']);
     }
 
+    public function testCreateAsyncRecord()
+    {
+        $this->testModel->id = 'id1';
+        $this->testModel->id2 = str_random(36);
+        $this->testModel->name = 'Test Create Async';
+        $this->testModel->count = 1;
+        $promise = $this->testModel->saveAsync();
+        \GuzzleHttp\Promise\all($promise)->wait();
```

Comment:
Don't think we need to wrap it in `all()`, just `$promise->wait()` should work. Better yet: `$this->testModel->saveAsync()->wait()`. 

Same comment for other places using `all`

Reply:
Makes sense. But for the case where there are multiple promises and user intends to resolve all of them together, then they would have to use `all`